### DISCUSSION TITLE
AdGuardHome (snap) with HAproxy (reverse).

### DISCUSSION THREAD

**Discussion by samasara-maker** [AdguardTeam/AdGuardHome#2837] [Q&A]
Good day,

I have installed Nextcloud as snap on a VPS server and now I want to install Adguarad as snap.
Now I started HAProxy as reverseproxy with this configuration: [Nextcloud-Snap Wiki.](https://github.com/nextcloud/nextcloud-snap/wiki/Putting-the-snap-behind-a-reverse-proxy#haproxy)
HAProxy is easy to understand but not for me.
My question: Is a wiki, with a sample configuration, possible for HAProxy?
Or can I get tips here for the right configuration?

Thanks for the work to Adguarad.

**Comment by ameshkov** [comment]
Tbh, I've never used HAProxy so it's not an easy question.

If someone can supply a sample configuration, we'd be happy to add it to FAQ.

**Comment by samasara-maker** [comment]
Am Mittwoch, den 17.03.2021, 20:18 +0000 schrieb Andrey Meshkov:
> Tbh, I've never used HAProxy so it's not an easy question.
> If someone can supply a sample configuration, we'd be happy to add it
> to FAQ.
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

Thanks for your mail,
-
I am considering building everything with Docker.
-
There are many, better and easy to find documentations (wikis) for
Docker.
-
Also how to use a reverseproxy is better described.
-
I will test it.
-
Maybe I will contact you about a Docker problem ;-)

**Comment by PolarDawn** [comment]
I'm having this problem. The nginx config works fine but I'm in a scenario were I have different proxies for different applications because some are easier to configure than others.

The problem is, requests to https://host:port/adguard/ is redirecting to https://host:port/login.html which isn't an address that works. If I go to https://host:port/adguard/login.html directly it works fine.

[use_backend adguard if { path /adguard }](url)

in the backend I have

```
backend adguard
        http-request replace-path /adguard(/)?(.*) /\2
        server adguard 127.0.0.1:3000
```

surprised nobody has had this problem

If I had adguard as the default with no extra path it works fine.
